cmake/templates/HPXMacros.cmake.in

```diff
@@ -93,11 +93,26 @@ function(hpx_setup_allocator)
     string(TOUPPER "${HPX_WITH_MALLOC}" HPX_MALLOC_UPPER)
     string(TOUPPER "${HPX_WITH_MALLOC_DEFAULT}" HPX_MALLOC_DEFAULT_UPPER)
     if(NOT (HPX_MALLOC_UPPER STREQUAL HPX_MALLOC_DEFAULT_UPPER))
-      hpx_error("The specified allocators do not match. This Project is configured with '${HPX_WITH_MALLOC}' and HPX was configured with '${HPX_WITH_MALLOC_DEFAULT}'.")
+      hpx_error("The specified allocators do not match. This Project is "
+        "configured with '${HPX_WITH_MALLOC}' and HPX was configured with "
+        "'${HPX_WITH_MALLOC_DEFAULT}'.")
     endif()
   endif()
   set(HPX_WITH_MALLOC ${HPX_WITH_MALLOC_DEFAULT} CACHE STRING
-    "Define which allocator should be linked in. Options are: system, tcmalloc, jemalloc, tbbmalloc, and custom (default is: tcmalloc)"
-    STRINGS "system;tcmalloc;jemalloc;mimalloc;tbbmalloc;custom")
+    "Define which allocator should be linked in. Options are: system, tcmalloc, "
+    "jemalloc, tbbmalloc, and custom (default is: tcmalloc)")
+  set_property(CACHE HPX_WITH_MALLOC
+    PROPERTY STRINGS "system;tcmalloc;jemalloc;mimalloc;tbbmalloc;custom")
```

Comment:
Could you explain what this change does?

Reply:
CMake allows to set a list of valid choices for an option (at least for the CMake GUI on Windows). The code I fixed represents this incantation...

![CMake GUI](https://user-images.githubusercontent.com/568063/68759336-18c5a380-05d5-11ea-9c03-f5457e1e0b9e.png)

Reply:
@msimberg: I have no idea why this fails however (https://circleci.com/gh/STEllAR-GROUP/hpx/142581) as the variable is set in the cache the line before CMake is complaining about it not being set in the cache :/

Reply:
Hmm, strange. I'll have a look.